core/src/Cabana_NeighborList.hpp

```diff
@@ -114,6 +116,50 @@ maxNeighbor( const ListType& list, const std::size_t num_particles )
 }
 } // namespace Impl
 
+/*!
+  \brief Construct a histogram of neighbors per particle.
+  \param list Neighbor list with valid NeighborList interface.
+  \param num_bin Number of bins for the histogram.
+  \return Neighbor list histogram.
+*/
+template <class ExecutionSpace, class ListType>
+Kokkos::View<int* [2], typename ListType::memory_space>
+neighborHistogram( ExecutionSpace exec_space, const std::size_t num_particles,
+                   const ListType& list, const int num_bin )
+{
+    // Allocate View of neighbors per particle
+    auto num_neigh = Kokkos::View<int*, typename ListType::memory_space>(
+        "particle_count", num_particles );
+
+    // Extract from neighbor list interface.
+    auto extract_functor = KOKKOS_LAMBDA( const int p )
+    {
+        num_neigh( p ) = NeighborList<ListType>::numNeighbor( list, p );
+    };
+    Kokkos::RangePolicy<ExecutionSpace> particle_policy( exec_space, 0,
+                                                         num_particles );
+    Kokkos::parallel_for( particle_policy, extract_functor );
+    Kokkos::fence();
+
+    auto bin_data = Cabana::binByKey( num_neigh, num_bin );
+
+    auto histogram = Kokkos::View<int* [2], typename ListType::memory_space>(
+        "particle_count", num_bin, 2 );
+    int max_neigh = NeighborList<ListType>::maxNeighbor( list );
```

Comment:
It looks like we call this on host here, but the `maxNeighbor` interface calls `numNeighbor` which accesses a device view, labelled `num_neighbors`, which leads to an inaccessible memory issue when using the neighbor histogram with HIP.